functional/vfio/run.sh

```diff
@@ -189,12 +210,21 @@ setup_configuration_file() {
 		if [ "$HYPERVISOR" = "qemu" ]; then
 			sed -i 's|^machine_type.*|machine_type = "'${MACHINE_TYPE}'"|g' "${kata_config_file}"
 			# Make sure we have set hot_plug_vfio to a reasonable value
-			sudo sed -i -e 's|^#hot_plug_vfio =.*$|hot_plug_vfio = "bridge-port"|' -e 's|^hot_plug_vfio = .*$|hot_plug_vfio = "bridge-port"|' "${kata_config_file}"
+			sed -i -e 's|^#hot_plug_vfio =.*$|hot_plug_vfio = "bridge-port"|' -e 's|^hot_plug_vfio =.*$|hot_plug_vfio = "bridge-port"|' "${kata_config_file}"
 		else
 			warn "Variable machine_type only applies to qemu. It will be ignored"
 		fi
 	fi
 
+        if [ "${VFIO_HOTPLUG}" != "bridge-port" ]; then
+		sed -i -e "s|^#?hot_plug_vfio =.*$|hot_plug_vfio = \"${VFIO_HOTPLUG}\"|"  -e "s|^hot_plug_vfio =.*$|hot_plug_vfio = \"${VFIO_HOTPLUG}\"|" "${kata_config_file}"
+        fi
+
+        if [ "${VFIO_COLDPLUG}" != "bridge-port" ]; then
+		sed -i -e "s|^#?cold_plug_vfio =.*$|cold_plug_vfio = \"${VFIO_COLDPLUG}\"|"  -e "s|^cold_plug_vfio =.*$|cold_plug_vfio = \"${VFIO_COLDPLUG}\"|" "${kata_config_file}"
+        fi
+
+
```

Comment:
Turns out this pattern
```
^#?hot_plug_vfio =.*$
```
doesn't work unless you pass `-E` to `sed. The coldplug test seems to have been skipped (judging from the jenkins log).

So how about you make it: `#*.*hot_plug_vfio.*`, and you can really collapse both cases:
```bash
$ cat test
#hot_plug_vfio = bla
# hot_plug_vfio = bla
hot_plug_vfio = bla
$ sed -e 's|^#*.*hot_plug_vfio.*|hot_plug_vfio = noport|' test
hot_plug_vfio = noport
hot_plug_vfio = noport
hot_plug_vfio = noport
```

Reply:
Ack makes sense, given this test file
```
hot_plug_vfio = "bridge-psdfasdfaort"
hot_plug_vfio="blub-blub"
hot_plug_vfio= "blbe"
hot_plug_vfio ="blfasdf"

# hot_plug_vfio = "bridge-psdfasdfaort"
# hot_plug_vfio="blub-blub"
# hot_plug_vfio= "blbe"
# hot_plug_vfio ="blfasdf"


#hot_plug_vfio = "bridge-psdfasdfaort"
#hot_plug_vfio="blub-blub"
#hot_plug_vfio= "blbe"
#hot_plug_vfio ="blfasdf"
```
and 
```
sed -e 's|^#*.*hot_plug_vfio.*|hot_plug_vfio = "no-port"|'
```
we get 
```
hot_plug_vfio = "no-port"
hot_plug_vfio = "no-port"
hot_plug_vfio = "no-port"
hot_plug_vfio = "no-port"

hot_plug_vfio = "no-port"
hot_plug_vfio = "no-port"
hot_plug_vfio = "no-port"
hot_plug_vfio = "no-port"


hot_plug_vfio = "no-port"
hot_plug_vfio = "no-port"
hot_plug_vfio = "no-port"
hot_plug_vfio = "no-port"
```